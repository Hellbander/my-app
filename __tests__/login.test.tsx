import '@testing-library/jest-dom'
import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { useRouter } from 'next/router'
import Login from '@/app/page'

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      prefetch: () => null
    }
  }
}))

describe('Login page', () => {
  it('renders login page', () => {
    const { container } = render(<Login />)
    expect(container).toMatchSnapshot()
  })

  it('should submit form', () => {
    const handleOnSubmitMock = jest.fn();
    render(<Login />)
    screen.getByRole("form", { name: 'login' }).onsubmit = handleOnSubmitMock
    fireEvent.click(screen.getByRole('button'))
    expect(handleOnSubmitMock).toHaveBeenCalledTimes(1);
  })

  it('should put email information', () => {
    render(<Login />)
    const input = screen.getByPlaceholderText('Enter your email here')
    fireEvent.change(input, {
      target: {
        value: 'email@email.com'
      }
    })
    expect(input).toHaveValue('email@email.com')
  })

  it('should put password information', () => {
    render(<Login />)
    const input = screen.getByPlaceholderText('Enter your password here')
    fireEvent.change(input, {
      target: {
        value: '12345678'
      }
    })
    expect(input).toHaveValue('12345678')
  })
})