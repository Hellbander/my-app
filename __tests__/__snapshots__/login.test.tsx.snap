// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders login page 1`] = `
<div>
  <main
    class="h-full flex justify-center align-middle"
  >
    <div
      class="flex flex-1 flex-col justify-center px-6 py-12 lg:px-8"
    >
      <div
        class="sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <h2
          class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
        >
          Sign in to your account
        </h2>
      </div>
      <div
        class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <form
          class="space-y-6"
          name="login"
        >
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="email"
            >
              Email address
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 p-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="email"
                name="email"
                placeholder="Enter your email here"
                type="text"
                value=""
              />
              <label
                class="block pt-1 text-xs font-normal leading-6 text-red-500"
              />
            </div>
          </div>
          <div>
            <div
              class="flex items-center justify-between"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="password"
              >
                Password
              </label>
            </div>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 p-2 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="password"
                name="password"
                placeholder="Enter your password here"
                type="password"
                value=""
              />
              <label
                class="block pt-1 text-xs font-normal leading-6 text-red-500"
              />
            </div>
          </div>
          <div>
            <button
              class="flex w-full justify-center rounded-md bg-blue-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-blue-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-600"
              id="sign-in"
              type="submit"
            >
              Sign in
            </button>
          </div>
        </form>
      </div>
    </div>
  </main>
</div>
`;
